src/browser/BrowserFunc.ts

```diff
@@ -158,7 +158,7 @@ export default class BrowserFunc {
             if (data.morePromotions?.length) {
                 data.morePromotions.forEach(x => {
                     // Only count points from supported activities
-                    if (['quiz', 'urlreward'].includes(x.promotionType)) {
+                    if (['quiz', 'urlreward'].includes(x.promotionType) && !x.attributes.is_unlocked) {
```

Comment:
Not sure if this has been rolled out for everyone, when inspecting my rewards html for the "is_unlocked" property, it does not appear to be anywhere, having it like this can cause errors for those who do not have this rolled out.

Reply:
If no modifications are made, the initial retrieval of available points will be incorrect because it includes locked activities, which cannot be completed.
Furthermore, when encountering locked activities in `More Promotions`, such as `The art of Cézanne` as shown in the image, errors may occur as follows.
The same issue is described in https://github.com/TheNetsky/Microsoft-Rewards-Script/issues/115 and https://github.com/TheNetsky/Microsoft-Rewards-Script/issues/130#issuecomment-2294862564.

```
[2024/8/18 00:21:23] [GO-HOME] Visited homepage successfully
[2024/8/18 00:21:24] [MAIN-POINTS] Current point count: 259
[2024/8/18 00:21:26] [MAIN-POINTS] You can earn 190 points today
[2024/8/18 00:22:17] [DAILY-SET] All "Daily Set" items have been completed
[2024/8/18 00:22:17] [MORE-PROMOTIONS] Started solving "More Promotions" items
[2024/8/18 00:22:36] [ACTIVITY] Found activity type: "UrlReward" title: "The art of Cézanne"
[2024/8/18 00:22:36] [URL-REWARD] Trying to complete UrlReward
[2024/8/18 00:22:36] [URL-REWARD] Completed the UrlReward successfully
[2024/8/18 00:22:55] [ACTIVITY] Found activity type: "UrlReward" title: "Block by block"
[2024/8/18 00:22:55] [URL-REWARD] Trying to complete UrlReward
[2024/8/18 00:22:55] [URL-REWARD] Completed the UrlReward successfully
[2024/8/18 00:22:58] [GET-NEW-TAB] Unable to get latest tab
[2024/8/18 00:22:58] [GET-NEW-TAB] An error occurred:undefined
[2024/8/18 00:22:58] [ACTIVITY] An error occurred:undefined
[2024/8/18 00:22:58] [GET-NEW-TAB] Unable to get latest tab
[2024/8/18 00:22:58] [GET-NEW-TAB] An error occurred:undefined
node:internal/process/promises:289
            triggerUncaughtException(err, true /* fromPromise */);
            ^

[UnhandledPromiseRejection: This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). The promise rejected with the reason "undefined".] {
  code: 'ERR_UNHANDLED_REJECTION'
}

Node.js v20.12.2
```

After setting `headless` to `false`, I found that it could not click on the `The art of Cézanne` activity link, resulting in subsequent errors that closed all tabs.

For my improved code, testing on 5 accounts has shown no issues, with 3 being level 2 and 2 being level 1.
and the logs
```
[2024/8/18 00:47:23] [MAIN-POINTS] Current point count: 259
[2024/8/18 00:47:25] [MAIN-POINTS] You can earn 60 points today
[2024/8/18 00:47:34] [GO-HOME] Visited homepage successfully
[2024/8/18 00:48:13] [DAILY-SET] All "Daily Set" items have been completed
[2024/8/18 00:48:13] [MORE-PROMOTIONS] All "More Promotion" items have already been completed
[2024/8/18 00:48:13] [PUNCH-CARD] All "Punch Cards" have already been completed
[2024/8/18 00:48:13] [SEARCH-BING] Starting bing searches
```

Locked activities look like this:

```
{
    name: 'ENstar_Rewards_locked_level2_Aug24w2_offer1',
    priority: -1,
    attributes: {
      animated_icon: '//az15297.vo.msecnd.net/images/rewards/membercenter/missions/Animated-Icons/paint.json',
      bg_image: '//az15297.vo.msecnd.net/images/rewards/membercenter/missions/Mobile/Dark-purple-color-75x75px.png',
      complete: 'False',
      description: 'View the groundbreaking artwork of Paul C茅zanne',
      destination: 'https://rewards.bing.com/levelbenefitexclusive',
      icon: '//az15297.vo.msecnd.net/images/rewards/membercenter/missions/Animated-Icons/Paint-3840x64px.png',
      image: '//az15297.vo.msecnd.net/images/rewards/membercenter/missions/278x180/Star-dark-purple-278x180px.png',
      link_text: 'See the gallery',
      max: '15',
      offerid: 'ENstar_Rewards_locked_level2_Aug24w2_offer1',
      progress: '0',
      sc_bg_image: '//az15297.vo.msecnd.net/images/rewards/membercenter/missions/Mobile/Dark-purple-color-75x75px.png',
      sc_bg_large_image: '//az15297.vo.msecnd.net/images/rewards/membercenter/missions/Mobile/Dark-purple-color-75x75px.png',
      small_image: '//az15297.vo.msecnd.net/images/rewards/membercenter/missions/Mobile/shoppingCart_75x75.png',
      state: 'Default',
      title: 'The art of C茅zanne',
      type: 'urlreward',
      give_eligible: 'True',
      user_level: 'level1',
      is_unlocked: 'False',
      locked_category_criteria: 'level2'
    },
    offerId: 'ENstar_Rewards_locked_level2_Aug24w2_offer1',
    complete: false,
    counter: 0,
    activityProgress: 0,
    activityProgressMax: 0,
    pointProgressMax: 15,
    pointProgress: 0,
    promotionType: 'urlreward',
    promotionSubtype: '',
    title: 'The art of C茅zanne',
    extBannerTitle: '',
    titleStyle: '',
    theme: '',
    description: 'View the groundbreaking artwork of Paul C茅zanne',
    extBannerDescription: '',
    descriptionStyle: '',
    showcaseTitle: '',
    showcaseDescription: '',
    imageUrl: '//az15297.vo.msecnd.net/images/rewards/membercenter/missions/278x180/Star-dark-purple-278x180px.png',
    dynamicImage: '',
    smallImageUrl: '//az15297.vo.msecnd.net/images/rewards/membercenter/missions/Mobile/shoppingCart_75x75.png',
    backgroundImageUrl: '//az15297.vo.msecnd.net/images/rewards/membercenter/missions/Mobile/Dark-purple-color-75x75px.png',
    showcaseBackgroundImageUrl: '//az15297.vo.msecnd.net/images/rewards/membercenter/missions/Mobile/Dark-purple-color-75x75px.png',
    showcaseBackgroundLargeImageUrl: '//az15297.vo.msecnd.net/images/rewards/membercenter/missions/Mobile/Dark-purple-color-75x75px.png',
    promotionBackgroundLeft: '',
    promotionBackgroundRight: '',
    iconUrl: '//az15297.vo.msecnd.net/images/rewards/membercenter/missions/Animated-Icons/Paint-3840x64px.png',
    animatedIconUrl: '//az15297.vo.msecnd.net/images/rewards/membercenter/missions/Animated-Icons/paint.json',
    animatedLargeBackgroundImageUrl: '',
    destinationUrl: 'https://rewards.bing.com/levelbenefitexclusive',
    linkText: 'See the gallery',
    hash: '8c91cc4bb848ab2239a485b66c013f663b44b03561f8bf789a9a9e47f7ab0863',
    activityType: '11',
    isRecurring: false,
    isHidden: false,
    isTestOnly: false,
    isGiveEligible: true,
    level: '',
    slidesCount: 0,
    legalText: '',
    legalLinkText: '',
    deviceType: '',
    exclusiveLockedFeatureType: 'level2',
    exclusiveLockedFeatureStatus: 'locked'
  },
```

Normal activities look like this:

```
{
    name: 'ZHCN_Rewards_Campaign_Global_Expansion2_BingOffer2',
    priority: 1,
    attributes: {
      animated_icon: '//az15297.vo.msecnd.net/images/rewards/membercenter/missions/Animated-Icons/soccerBall.json',
      bg_image: '//az15297.vo.msecnd.net/images/rewards/membercenter/missions/Animated-Banners/Banner_Rewards__small.json',
      complete: 'True',
      description: 'Do you think this soccer player is the best of all time?',
      destination: 'https://www.bing.com/search?q=Lionel+Messi&form=ML23ZW',
      icon: '//az15297.vo.msecnd.net/images/rewards/membercenter/missions/Animated-Icons/soccerBall.json',
      image: '//az15297.vo.msecnd.net/images/rewards/membercenter/missions/278x180/Star-magenta-278x180px.png',
      link_text: 'Learn more',
      max: '5',
      offerid: 'ZHCN_Rewards_Campaign_Global_Expansion2_BingOffer2',
      progress: '5',
      promotional: 'True',
      sc_bg_image: '//az15297.vo.msecnd.net/images/rewards/membercenter/missions/Animated-Banners/Banner_Rewards__medium.json',
      sc_bg_large_image: '//az15297.vo.msecnd.net/images/rewards/membercenter/missions/Animated-Banners/Banner_Rewards__large.json',
      small_image: '//az15297.vo.msecnd.net/images/rewards/membercenter/missions/Mobile/soccerBall_75x75.png',
      state: 'Complete',
      title: 'Soccer legend',
      type: 'urlreward',
      give_eligible: 'True'
    },
    offerId: 'ZHCN_Rewards_Campaign_Global_Expansion2_BingOffer2',
    complete: true,
    counter: 0,
    activityProgress: 0,
    activityProgressMax: 0,
    pointProgressMax: 5,
    pointProgress: 5,
    promotionType: 'urlreward',
    promotionSubtype: '',
    title: 'Soccer legend',
    extBannerTitle: '',
    titleStyle: '',
    theme: '',
    description: 'Do you think this soccer player is the best of all time?',
    extBannerDescription: '',
    descriptionStyle: '',
    showcaseTitle: '',
    showcaseDescription: '',
    imageUrl: '//az15297.vo.msecnd.net/images/rewards/membercenter/missions/278x180/Star-magenta-278x180px.png',
    dynamicImage: '',
    smallImageUrl: '//az15297.vo.msecnd.net/images/rewards/membercenter/missions/Mobile/soccerBall_75x75.png',
    backgroundImageUrl: '//az15297.vo.msecnd.net/images/rewards/membercenter/missions/Animated-Banners/Banner_Rewards__small.json',
    showcaseBackgroundImageUrl: '//az15297.vo.msecnd.net/images/rewards/membercenter/missions/Animated-Banners/Banner_Rewards__medium.json',
    showcaseBackgroundLargeImageUrl: '//az15297.vo.msecnd.net/images/rewards/membercenter/missions/Animated-Banners/Banner_Rewards__large.json',
    promotionBackgroundLeft: '',
    promotionBackgroundRight: '',
    iconUrl: '//az15297.vo.msecnd.net/images/rewards/membercenter/missions/Animated-Icons/soccerBall.json',
    animatedIconUrl: '//az15297.vo.msecnd.net/images/rewards/membercenter/missions/Animated-Icons/soccerBall.json',
    animatedLargeBackgroundImageUrl: '',
    destinationUrl: 'https://www.bing.com/search?q=Lionel+Messi&form=ML23ZW',
    linkText: 'Learn more',
    hash: '98ad518e6d45d6a4ac3a3e7809d4474aa68e8ff8da1430394f1bf2de56df962a',
    activityType: '11',
    isRecurring: false,
    isHidden: false,
    isTestOnly: false,
    isGiveEligible: true,
    level: '',
    slidesCount: 0,
    legalText: '',
    legalLinkText: '',
    deviceType: '',
    exclusiveLockedFeatureType: 'notsupported',
    exclusiveLockedFeatureStatus: 'notsupported'
  },
```

Reply:
With this change, it will be locked the eligible newer promotions for ALL users, because the param is always `true`. I suggest checking the `x.exclusiveLockedFeatureStatus` value if it is 'unlocked', so eligible users can claim it without any problems.